src/app/parser.rs

```diff
@@ -1790,6 +1790,38 @@ impl<'a, 'b> Parser<'a, 'b>
         }
         Ok(())
     }
+    
+    pub fn add_from_env(&mut self, matcher: &mut ArgMatcher<'a>) -> ClapResult<()> {
+        macro_rules! add_val {
```

Comment:
This is duplicated from `add_defaults`, I'll make sure to dedup before final submission.